sklearn/base.py

```diff
@@ -437,8 +437,20 @@ def _validate_data(self, X, y=None, reset=True,
 
         return out
 
+    @property
```

Comment:
Maybe add a comment to explain that `@property` is necessary to make `hasattr(estimator, "_repr_html_")` return `True` or `False` depending on `get_config()["display"] `.